Add support for excluding fmt.Fprint* to a buffer type

yasushi-saito opened the pull request:
For example,

b := bytes.Buffer{}
fmt.Fprintf(&b, "%d", 10)

should be exempt from errchecks, just like fmt.Printf already is.

This change also exempts fmt.Fprint*(os.Stderr, ....) for the same reason.

kisielk commented:
Interesting idea. I kind of like where this is going.
Wouldn't it make more sense to pass the type of the first argument instead of an arbitrary string like BUFFER?
So you could write: `fmt.Fprintf(*bytes.Buffer)` to ignore calls to that function where the first argument is `*bytes.Buffer`? It's a bit more verbose and would result in more rules, but it would allow the system to be extended to arbitrary types.

kisielk commented:
It would still need a special string for eg: STDOUT and STDERR though.

yasushi-saito commented:
Thanks, changed the exclude format as you suggest.

kisielk commented:
Looking good. Do you mind adding a few tests as well?

yasushi-saito commented:
I added a new testdata, testdata/main3.go. I also modified the main_test a bit. It wasn't checking that errcheck wouldn't produce an unchecked errors. Let me know if you want something different.

yasushi-saito commented:
Gentle ping. The test errors looks like just unrelated incompatibility w/ older versions of go (can we disable these versions?)

kisielk commented:
I'm not sure I'm ready to drop support for Go 1.9 yet. Can you adjust the tests so the portions with the `strings.Builder` only compile on 1.10 and above?
